doc/CodeStyle.md

```diff
@@ -0,0 +1,217 @@
+# Introduction #
+
+As in any open source project, many people will read and modify the code. Quite likely this will happen long after you have left. Therefore, it is important to make the code easily understandable.
+
+To make this happen, the project has a number of rules and guide lines for code.
+  * Document every part.
+  * Use algorithms that scale well without being cryptic.
+  * Use the code style explained below.
+
+## Documenting code ##
+
+Everybody knows we should do it, but most programmers think they can live without. When you write programs of less than a thousand lines that you never give to other people, you probably can.
+FreeRCT is however close to 75,000 lines (at time of writing) and growing. There are just too many details to remember all.
+
+Of course you can read the code of the method to understand what it does, and for code that you want to change you have to do that, but what if you just want to call the function?
+The latter kind of use is where documentation makes the difference. In this project, you can read just a few lines of text instead of reverse engineering the intended use from the code, and hope you didn't miss an important detail.
+(Alberth: Since I switched to documenting all my code, I found that it actually speeds up my work instead of slowing down, despite the additional time needed to write the documentation.)
```

Comment:
Make it a bit more clear it's a quote?

( Quote by Alberth: ``Since ... documentation.'' )